crates/yttrium/src/chain_abstraction/client.rs

```diff
@@ -396,13 +397,66 @@ impl Client {
         }
     }
 
-    // pub async fn send() {
-    //     // TODO input signed txns
-    //     // TODO send route first, etc. (possibly in parallel)
-    //     // TODO wait_for_success
-    //     // TODO send initial transaction
-    //     // TODO await initial transaction success
-    // }
+    pub async fn execute(
+        &self,
+        ui_fields: UiFields,
```

Comment:
why do we nee `ui_fields`?

Reply:
We need more than the signatures, we need the original transactions too in order to construct the signed transaction to send to the chain. All the txns are contained in `UiFields` so IMO it's easier to just pass the result from `prepare_detailed()` back into `execute()` than to bother/risk the implementor doing it all.

We also need `orchestration_id` and `metadata.check_in`, and it's better to not risk the implementor messing these parameters up. There might be other fields we want in the future too.